### ISSUE TITLE
wav2vec 2.0: L2 penalty on features

### ISSUE THREAD

**Issue by yuanenming** [pytorch/fairseq#3315]
#### What is your question?
The Wav2vec 2.0 paper just mentioned two losses, namely, contrastive loss and diversity loss. But in the implementation, I see there is also a `features_pen` loss, which is the L2 norm of features extracted from CNN. Moreover, this loss has a very large weight parameter (10, while only 0.1 for diversity loss) when pretraining.

So my question is,
What’s the use of this `features_pen` loss?
Why it has such a large weight?
Is there any ablation study about this loss?

@alexeib

**Comment by alexeib** [comment]
this parameter is mostly to stabilize a model that uses the "default" extractor mode (i.e. prevent it from crashing late in training). it has very little effect on final accuracy. for new models, i recommend using task.normalize: true, extractor_mode: layer_norm and set the penalty weight for this parameter to 0 (i.e. disable it)

**Comment by alexeib** [comment]
feature magnitude i think is mentioned in the footnote. that is just to stabilize training when using the default feature extractor. you can switch to task.normalize=true, model.feature_extractor_mode='layer_norm' and then set this coefficient to 0 and it will work fine. this is the configuration i recommend for new models
